src/bigint.rs

```diff
@@ -878,6 +882,45 @@ impl BigUint {
     }
 
     /// Creates and initializes a `BigUint`.
+    ///
+    /// The bytes are in big-endian byte order
+    ///
+    /// # Examples
+    ///
+    /// ```
+    /// use num::bigint::BigUint;
+    ///
+    /// assert_eq!(BigUint::from_bytes("A".as_bytes()),
+    ///            BigUint::parse_bytes("65".as_bytes(), 10));
+    /// assert_eq!(BigUint::from_bytes("AA".as_bytes()),
+    ///            BigUint::parse_bytes("16705".as_bytes(), 10));
+    /// assert_eq!(BigUint::from_bytes("AB".as_bytes()),
+    ///            BigUint::parse_bytes("16706".as_bytes(), 10));
+    /// assert_eq!(BigUint::from_bytes("Hello world!".as_bytes()),
+    ///            BigUint::parse_bytes("22405534230753963835153736737".as_bytes(), 10));
+    /// ```
+    #[inline]
+    pub fn from_bytes(msg: &[u8]) -> Option<BigUint> {
+        BigUint::parse_bytes(msg.to_hex().as_bytes(), 16)
+    }
+
+    /// Returns the byte representation of the `BigUint`
+    ///
+    /// The bytes are in big-endian byte order
+    #[inline]
+    pub fn to_bytes(&self) -> Vec<u8> {
+        let mut a = self.clone();
+        let mut result = Vec::new();
+        let byte_size = ToBigUint::to_biguint(&256).unwrap();
+        while !a.is_zero() {
```

Comment:
I think this can be far more efficient, at the very least it could use shifts and masks instead of division and modulus, but, even better: 256 divides the base (2<sup>32</sup>), so the conversion can be done word-by-word, by iterating over `self.data` (in reverse, I guess).